contracts/src/interfaces/IHyperdriveMatchingEngine.sol

```diff
@@ -0,0 +1,168 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.20;
+
+import { IMorpho } from "morpho-blue/src/interfaces/IMorpho.sol";
+import { IMorphoFlashLoanCallback } from "morpho-blue/src/interfaces/IMorphoCallbacks.sol";
+import { IHyperdrive } from "../interfaces/IHyperdrive.sol";
+
+/// @title IHyperdriveMatchingEngine
+/// @notice Interface for the Hyperdrive matching engine.
+interface IHyperdriveMatchingEngine is IMorphoFlashLoanCallback {
+    /// @notice Thrown when an order is already cancelled.
+    error AlreadyCancelled();
+
+    /// @notice Thrown when an order is already expired.
+    error AlreadyExpired();
+
+    /// @notice Thrown when the destination for the add or remove liquidity
+    ///         options isn't configured to this contract.
+    error InvalidDestination();
+
+    /// @notice Thrown when orders that don't cross are matched.
+    error InvalidMatch();
+
+    /// @notice Thrown when the order type doesn't match the expected type.
+    error InvalidOrderType();
+
+    /// @notice Thrown when an address that didn't create an order tries to
+    ///         cancel it.
+    error InvalidSender();
+
+    /// @notice Thrown when `asBase = false` is used. This implementation is
+    ///         opinionated to keep the implementation simple.
+    error InvalidSettlementAsset();
+
+    /// @notice Thrown when the signature for an order intent doesn't recover to
+    ///         the expected signer address.
+    error InvalidSignature();
+
+    /// @notice Thrown when the long and short orders don't refer to the same
+    ///         Hyperdrive instance.
+    error MismatchedHyperdrive();
```

Comment:
NIT: Maybe move these to the end? `IHyperdrive` isn't a great reference since it's so broken up, but a common order for interfaces I've seen elsewhere seems to be:
```
events

structs

stateful fn's

getter fn's

errors
```

Reply:
Okay awesome! Not a huge deal and didn't really have a great source of truth on this one, solidity docs to the rescue!